### ISSUE TITLE
Interpretation of return values from hb_ot_layout_get_baseline

### ISSUE THREAD

**Issue by matthiasclasen** [harfbuzz/harfbuzz#3447]
The OpenType spec says this about the BASE table: 

    Each baseline coordinate is defined as a single X or Y value in design units measured from the zero position on the relevant X or Y axis.

But it doesn't actually say what the zero position for the axes is.

It would be plausible to use the origin of vertical glyph metrics for vertical baselines (which would put the central baseline at 0).

**Comment by behdad** [comment]
Right. The more I think about it, the more it makes sense that for vertical, it's relative to the vertical origin indeed; that is, the central baseline.

**Comment by matthiasclasen** [comment]
May need to look at some actual font BASE tables which have vertical baselines

**Comment by matthiasclasen** [comment]
Here is what we are seeing in Noto Sans CJK for vertical baselines:

```
ideo: 0
icfb: 46
icft: 954
```

so it seems that these values are in design coordinates, with an embox that goes from 0 to 1000.

**Comment by behdad** [comment]
> Here is what we are seeing in Noto Sans CJK for vertical baselines:
> 
> ```
> ideo: 0
> icfb: 46
> icft: 954
> ```
> 
> so it seems that these values are in design coordinates, with an embox that goes from 0 to 1000.

Okay this is completely unusable, given that each glyph can have a different vertical origin. I don't think anyone can sanely use these...

**Comment by matthiasclasen** [comment]
Here is another problem report about (horizontal) icfb: https://gitlab.gnome.org/GNOME/pango/-/issues/675

**Comment by punchcutter** [comment]
I'm a little confused about some of these comments. The Noto CJK values for VertAxis are on the x-axis and have nothing to do with Vertical Origin which is on the y-axis. 'ideo' being set to 0 for vertical text is stated explicitly in the [spec](https://docs.microsoft.com/en-us/typography/opentype/spec/baselinetags):

> Ideographic em-box left edge. If this tag is present in the VertAxis, the value must be set to 0. (See [Ideographic Em-Box](https://docs.microsoft.com/en-us/typography/opentype/spec/baselinetags#ideoembox) below for usage.)

As for the zero position I thought that's also pretty clear: 

> Metrics in the BASE table are expressed directly in BaseCoord tables using explicit X or Y font-unit values. In a variable font, these X and Y values apply to the default instance and may need to be adjusted for the current variation instance.

**Comment by behdad** [comment]
> The Noto CJK values for VertAxis are on the x-axis and have nothing to do with Vertical Origin which is on the y-axis.

I meant vertical origin of glyphs, which has an x component half of the glyph advance. For CJK glyphs this is the same for all glyphs. But for non-CJK glyphs it differs...

**Comment by punchcutter** [comment]
Got it. That isn't so clear.
